File: hanoi/Hanoi.java
public class Hanoi {
	private static void hanoi(int n, int a, int b) {
		int temp;
		if (n != 1) {
			temp = 6 - a - b;
			hanoi(n - 1, a, temp);
			hanoi(n - 1, temp, b);
		}
	}

	public static void main(String[] args) {
		long st = System.currentTimeMillis();
		hanoi(Integer.parseInt(args[0]), 1, 2);
		long t = System.currentTimeMillis() - st;
		System.out.println(t + " ms");
	}
}

File: hanoi/hanoi.php
<?php
  function hanoi($n, $a, $b) {
    if($n!=1) {
      $temp=6-$a-$b;
      hanoi($n-1, $a, $temp);
      hanoi($n-1, $temp, $b);
    }
  }

  $st = microtime(true);
  hanoi(20, 1, 2);
  $t = round((microtime(true) - $st) * 1000);
  echo $t." ms\n";
?>
